packages/arb-ts/integration_test/arb-bridge.test.ts

```diff
@@ -385,14 +386,172 @@ describe('ERC20', () => {
-  //     l1TestWallet.address
-  //   )
-  //   console.log(l1EventData.length, withdrawTokenData.length)
-  //   expect(l1EventData.length).to.equal(withdrawTokenData.length)
-  // })
+})
+
+describe('CustomToken', () => {
+  let l1CustomToken: TestCustomTokenL1
+  let l2CustomToken: TestArbCustomToken
+  if (!existantCustomTokenL1 && !existantCustomTokenL2) {
+    it('sets up a new custom token, L1 and L2', async () => {
+      prettyLog("No custom token addresses given; we'll do it live!")
+      const customTokenFactory = await new TestCustomTokenL1__factory(
+        preFundedWallet
+      )
+      l1CustomToken = await customTokenFactory.deploy(
+        bridge.ethERC20Bridge.address
+      )
+      let rec = await l1CustomToken.deployTransaction.wait()
+      expect(rec.status).to.equal(1)
+      prettyLog('Deployed a new customL1 Token at ' + l1CustomToken.address)
+
+      const customL2TokenFactory = await new TestArbCustomToken__factory(
+        l2TestWallet
+      )
+      l2CustomToken = await customL2TokenFactory.deploy(
+        bridge.arbTokenBridge.address,
+        l1CustomToken.address
+      )
+      rec = await l2CustomToken.deployTransaction.wait()
+      expect(rec.status).to.equal(1)
+      prettyLog('Deployed a new custom L2 token at ' + l2CustomToken.address)
+    })
+    it('l1 token registers custom L2 token and notifies the ArbTokenBridge', async () => {
+      const registerRes = await l1CustomToken.registerTokenOnL2(
+        l2CustomToken.address,
+        Zero,
+        BigNumber.from(10000000000000),
+        Zero,
+        l1TestWallet.address
+      )
+      const registerRec = await registerRes.wait()
+      expect(registerRec.status).to.equal(1)
+
+      const eventData = (
+        await BridgeHelper.getActivateCustomTokenEventResult(
+          registerRec,
+          bridge.ethERC20Bridge.address
+        )
+      )[0]
+
+      expect(eventData).to.exist
+
+      const { seqNum } = eventData
+
+      const l2RetriableHash = await bridge.calculateL2RetryableTransactionHash(
+        seqNum
+      )
+
+      const retriableReceipt = await arbProvider.waitForTransaction(
+        l2RetriableHash
+      )
+
+      expect(retriableReceipt.status).to.equal(1)
+
+      wait()
+      const l2AddressHopefully = await bridge.arbTokenBridge.customToken(
+        l1CustomToken.address
+      )
+      expect(l2AddressHopefully).to.equal(l2CustomToken.address)
+    })
+  } else {
+    l1CustomToken = TestCustomTokenL1__factory.connect(
+      existantCustomTokenL1,
+      preFundedWallet
+    )
+    l2CustomToken = TestArbCustomToken__factory.connect(
+      existantCustomTokenL2,
+      l2TestWallet
+    )
+    prettyLog(
+      `Connected to pre-deployed custom token addresses. L1:${l1CustomToken.address} L2: ${l2CustomToken.address}`
+    )
+  }
+
+  it('setup: mint some custom token and send to test address', async () => {
+    const mintRes = await l1CustomToken.mint()
```

Comment:
Might seem a bit useless since its only for testing, but for the future it might be useful to inject the amount to mint as a parameter. Then we don't need to hardcode the `40000000` a couple of lines below